dev-db/mongodb/mongodb-8.0.12.ebuild

```
@@ -0,0 +1,213 @@
+	app-arch/zstd:=
+	>=dev-cpp/yaml-cpp-0.6.2:=
+	dev-libs/boost:=[nls]
+	>=dev-libs/libpcre-8.42[cxx]
+	dev-libs/snowball-stemmer:=
+	net-misc/curl
+	>=sys-libs/zlib-1.2.12:=
+	kerberos? ( dev-libs/cyrus-sasl[kerberos] )
+	ssl? (
+		>=dev-libs/openssl-1.0.1g:0=
+	)"
+DEPEND="${RDEPEND}
+	${PYTHON_DEPS}
+	sys-libs/ncurses:0=
+	sys-libs/readline:0=
+	debug? ( dev-debug/valgrind )"
+BDEPEND="
+	$(python_gen_any_dep '
+		>=dev-build/scons-3.1.1[${PYTHON_USEDEP}]
+		dev-python/cheetah3[${PYTHON_USEDEP}]
+		dev-python/psutil[${PYTHON_USEDEP}]
+		dev-python/pyyaml[${PYTHON_USEDEP}]
+		dev-python/distro[${PYTHON_USEDEP}]
+		dev-python/gitpython[${PYTHON_USEDEP}]
+		dev-python/poetry[${PYTHON_USEDEP}]
+		dev-python/pymongo[${PYTHON_USEDEP}]
+		dev-python/tenacity[${PYTHON_USEDEP}]
+	')
+"
+PDEPEND="
+	mongosh? ( app-admin/mongosh-bin )
+	tools? ( >=app-admin/mongo-tools-100 )
+"
+
+PATCHES=(
+	"${WORKDIR}/mongodb-8.0.8-patches/mongodb-4.4.29-no-enterprise.patch"
+	"${WORKDIR}/mongodb-8.0.8-patches/${PN}-5.0.2-no-compass.patch"
+	"${WORKDIR}/mongodb-8.0.8-patches/${PN}-8.0.6-skip-reqs-check.patch"
+	"${WORKDIR}/mongodb-8.0.8-patches/${PN}-8.0.6-fixes-for-boost-1.85.patch"
+	"${WORKDIR}/mongodb-8.0.8-patches/${PN}-8.0.4-scons.patch"
+	"${WORKDIR}/mongodb-8.0.8-patches/${PN}-8.0.6-use-tenacity.patch"
+	"${FILESDIR}/boost_issue_402.patch"
+	"${FILESDIR}/${PN}-8.0.12-sconstruct.patch"
+)
+
+python_check_deps() {
+	python_has_version -b ">=dev-build/scons-3.1.1[${PYTHON_USEDEP}]" &&
+	python_has_version -b "dev-python/cheetah3[${PYTHON_USEDEP}]" &&
+	python_has_version -b "dev-python/psutil[${PYTHON_USEDEP}]" &&
+	python_has_version -b "dev-python/pyyaml[${PYTHON_USEDEP}]"
+	python_has_version -b "dev-python/distro[${PYTHON_USEDEP}]"
+	python_has_version -b "dev-python/gitpython[${PYTHON_USEDEP}]"
+	python_has_version -b "dev-python/poetry[${PYTHON_USEDEP}]"
+	python_has_version -b "dev-python/pymongo[${PYTHON_USEDEP}]"
+	python_has_version -b "dev-python/tenacity[${PYTHON_USEDEP}]"
+}
+
+pkg_pretend() {
+	# Bug 809692 + 890294
+	if use amd64 && ! use cpu_flags_x86_avx; then
+		ewarn "MongoDB 5.0 requires use of the AVX instruction set."
+		ewarn "This ebuild will use --experimental-optimization=-sandybridge which"
+		ewarn "will result in an experimental build of MongoDB as per upstream."
+		ewarn "https://docs.mongodb.com/v5.0/administration/production-notes/"
+	fi
+
+	if [[ -n ${REPLACING_VERSIONS} ]]; then
+		if ver_replacing -lt 4.4; then
+			ewarn "To upgrade from a version earlier than the 4.4-series, you must"
+			ewarn "successively upgrade major releases until you have upgraded"
+			ewarn "to 4.4-series. Then upgrade to 5.0 series."
+		else
+			ewarn "Be sure to set featureCompatibilityVersion to 7.0 before upgrading."
+		fi
+	fi
+}
+
+src_prepare() {
+	default
+	# remove bundled libs
+	rm -r src/third_party/{boost,snappy,yaml-cpp} || die
+
+	# remove compass
+	rm -r src/mongo/installer/compass || die
+}
+
+src_configure() {
```

```suggestion
src_configure() {
	# bugs #954813 and #942551
	filter-lto
```

This would be nice as well. Did so for the other version as I just went through them all, funnily enough all build with gcc-15 :/
https://github.com/gentoo/gentoo/pull/43287

---

This changes nothing for 8.0.12. Still the same error:
  184 |     return Policy::value(&*try_emplace(key).first);
      |                            ~~~~~~~~~~~^~~~~
src/mongo/db/timeseries/bucket_catalog/bucket_catalog_internal.cpp:872:97:   required from here
  872 |         auto& [timeField, refCount] = stripe.collectionTimeFields[bucket.bucketId.collectionUUID];
      |                                                                                                 ^
/usr/lib/gcc/x86_64-pc-linux-gnu/15/include/g++-v15/bits/uses_allocator_args.h:71:29: error: static assertion failed: construction with an allocator must be possible if uses_allocator is true
   71 |               static_assert(is_constructible_v<_Tp, _Args..., const _Alloc&>,

---

The lto bit is for lto bugs. Its not going to help with gcc-15. 

It was an observation I had as I built them all.